src/net/sourceforge/kolmafia/persistence/AdventureDatabase.java

```diff
@@ -45,7 +44,8 @@ public class AdventureDatabase {
   public static final Map<String, String> PARENT_ZONES = new HashMap<>();
   public static final Map<String, String> ZONE_DESCRIPTIONS = new HashMap<>();
 
-  private static final StringArray[] adventureTable = new StringArray[4];
+  private static final List<ArrayList<String>> adventureTable =
+      List.of(new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
```

Comment:
If you don't like StringArray[4] (cool) then why not ArrayList<String>[4]?

The code, as exists, KNOWS there are indices 0-3, and you replaced adventureTable[0] with adventureTable.get(0). Why is that an improvement?

If you want to replace the fixed-size (4) array with something, why not make an object with 4 fields, or something, and replace the magic indices with fields names?